### PR TITLE
Fixes issue #885.

### PR DESCRIPTION
**Pull request by semajtrip**

### CODE REVIEW

**Comment by noah-** [comment]
Hi, nice diagnosis on the bug. I think a proper solution is to fix the open chest code to check target chest area, if area is not same as current area it should return to current area after doing open chest.

**Comment by noah-** [comment]
I don't think we need a "stuck counter" check, should always revert to "previous" if current area is different from previous area.

File: d2bs/kolbot/libs/common/Attack.js [hunk 1]
```diff
@@ -776,6 +782,8 @@ var Attack = {
 			room = rooms.shift();
 
 			result = Pather.getNearestWalkable(room[0], room[1], 18, 3);
+			
+			previous = result;
```
[the remark this answers is not in the document]
Comment (semajtrip) [reply d2bs/kolbot/libs/common/Attack.js]: Great catch. I moved this so it will be set only when result has been set.

**Comment by semajtrip** [comment]
Thanks for the great feedback! The area id will now only be checked when the result is false. I did more debugging and realized that when the bot goes out of the specified area Pather.getNearestWalkable always returns false which puts the function in an infinite loop.

File: d2bs/kolbot/libs/common/Attack.js [hunk 2]
```diff
@@ -779,12 +781,17 @@ var Attack = {
...
 				Pather.moveTo(result[0], result[1], 3, spectype);
+				previous = result;
 				//this.countUniques();
 
 				if (!this.clear(40, spectype)) {
 					break;
 				}
 			}
+			// Make sure bot does not get stuck in different area.
+			else if (currentArea !== getArea().id) {
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Attack.js]: Should this be else if or just if?
For example, if this.clear is successful, but results in player leaving the area (chasing mobs), we would still want it to return to previous location right?

Comment (semajtrip) [reply d2bs/kolbot/libs/common/Attack.js]: This should be an else if. Current area only needs to be checked when result is false. In the example when this.clear is successful the player may be stuck in another area. The next iteration of the while loop will calculate result as false then move the player back to the previous room.

**Comment by semajtrip** [comment]
I will submit pull requests for other branches once I figure out how to do it again =)

**Comment by noah-** [comment]
You have tested after making these changes? If so, I think it is ready to merge.

**Comment by semajtrip** [comment]
I have been running the bot, for several days, with this change without any issues. I have been botting a wide variety of areas including all of the major bosses. I also disabled the max game time to ensure the bot was not getting stuck at any point.

File: d2bs/kolbot/libs/common/Attack.js [hunk 3]
```diff
@@ -736,7 +736,7 @@ var Attack = {
 			say("clearlevel " + getArea().name);
 		}
 
-		var room, result, rooms, myRoom;
+		var room, result, rooms, myRoom, currentArea, previous;
```
Comment (noah-) [inline d2bs/kolbot/libs/common/Attack.js]: rename currentArea to current or previous to previousArea for consistency

**Pull request merged**